barretenberg/cpp/src/barretenberg/common/thread_utils.cpp

```diff
@@ -0,0 +1,20 @@
+#include "thread_utils.hpp"
+
+/**
+ * @brief calculates number of threads to create based on minimum iterations per thread
+ * @details Finds the number of cpus with get_num_cpus(), and calculates `desired_num_threads`
+ * Returns the min of `desired_num_threads` and `max_num_theads`.
+ * Note that it will not calculate a power of 2 necessarily, use `calc_num_threads_pow2` instead
```

Comment:
should be max_num_threads, also calc_num_threads_pow2 doesn't exist

Reply:
Not sure what you mean with these comments here. Did you intend to implement calc_num_threads_pow2?

Reply:
just a note to fix the typo of "max_num_theads" and won't add the pow2 function for now unless you think I should add it.

Reply:
Yes, I think you should add it because this utility function is nearly identical to whats used in Sumcheck so the code is just duplicated there (with `get_num_cpus_pow2` instead of `get_num_cpus`).

Reply:
But of course you should not assume that min_num_iterations is the same between those two very different use cases. In your case you're doing 1 field op. In Sumcheck its doing much more per iteration.

Reply:
Yeah defining a local variable and passing it as argument to the function is what I had in mind. Or the caller can pass nothing and use the default

Reply:
got it, ended up just using the default now